crypto/evp/digest.c

```diff
@@ -179,7 +179,7 @@ int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *size)
     return ret;
 }
 
-int EVP_DigestFinalXOF(EVP_MD_CTX *ctx, unsigned char *md, size_t size)
+int EVP_DigestSqueezeXOF(EVP_MD_CTX *ctx, unsigned char *md, size_t size)
```

Comment:
You should probably check in here that the flag `EVP_MD_CTX_FLAG_CLEANED` isn't already set.

Reply:
Say what?  You're setting that flag in the new `EVP_DigestFinalXOF`!

Reply:
Only when ctx->digest->cleanup is non-NULL.